kratos/modeler/operation/find_contacts_in_skin_model_part.cpp

```diff
@@ -0,0 +1,250 @@
+                {
+                    //This cell intersects with condition and has correct color
+                    const array_1d<std::size_t, 3> indexes{i,j,k};
+                    std::vector<array_1d<std::size_t, 3>> face_neigh_indexes;
+                    GetCellNeihgbourColorsConnectedByFace(indexes, r_colors, face_neigh_colors, face_neigh_indexes);
+
+                    // We check if there is any neighbour that satisfies:
+                    // is connected by face, projection is within certain angle, distance
+                    // to condition is minimum.
+                    GetClosestContactColorFromNeighbours(
+                        rContactContainer.mColorVector,
+                        face_neigh_colors,
+                        face_neigh_indexes,
+                        condition_center,
+                        condition_normal,
+                        r_colors,
+                        pairing_status,
+                        min_distance,
+                        temporary_color);
+
+                    // if not contact by face has been found we look for contacts by corner.
+                    if (pairing_status < 2) {
+                        const array_1d<std::size_t,3> indexes{i,j,k};
+                        GetCellNeihgbourColors(indexes,all_neigh_colors, r_colors);
+                        for (const auto& contact_color : rContactContainer.mColorVector) {
+                            if (std::find(all_neigh_colors.begin(), all_neigh_colors.end(),contact_color) != all_neigh_colors.end()) {
+                                pairing_status = 1;
+                                temporary_color = contact_color;
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    if (pairing_status > 0) {
+        //We did find a temporary candidate , so we use this one
+        KRATOS_CRITICAL_SECTION
+        {
+            rContactContainer.mConditionsMap[temporary_color].push_back(rCondition.Id());
+            for (const auto& r_node : rCondition.GetGeometry()) {
+                rContactContainer.mNodeMap[temporary_color].push_back(r_node.Id());
+            }
+        }
+    }
+}
+
+void FindContactsInSkinModelPart::GetClosestContactColorFromNeighbours(
+    const std::vector<int>& rContactColors,
+    const std::vector<int>& rNeighbourColors,
+    const std::vector<array_1d<std::size_t, 3>>& rNeighbourIndexes,
+    const array_1d<double,3>& rConditionCenter,
+    const array_1d<double,3>& rConditionNormal,
+    const CartesianMeshColors& rColors,
+    int& rPairingStatus,
+    double& rMinDistance,
+    int& rTemporaryColor) const
+{
+    constexpr double angle_threshold = 0.3;
+    int i = 0;
+    for(auto& ind: rNeighbourIndexes){
+        auto cell_center = rColors.GetCenterOfElement(ind[0], ind[1], ind[2]);
+        const int neigh_color = rNeighbourColors[i];
```

Comment:
Could use a `std::vector<int>::iterator rNeighbourColorsItr` and access to `*rNeighbourColorsItr` to avoid the auxiliary variable `i`